From: paul cannon <pcannon@epochlabs.com>
Date: Wed, 9 Aug 2017 17:46:39 -0500
Subject: Fix size calculations under send_trace_metadata

---
 cmd-recv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd-recv.c b/cmd-recv.c
index fb01794..49eb51d 100644
--- a/cmd-recv.c
+++ b/cmd-recv.c
@@ -168,7 +168,7 @@ void send_trace_metadata(int sock, const char *dirname, char *filename)
 	struct uftrace_msg msg = {
 		.magic = htons(UFTRACE_MSG_MAGIC),
 		.type  = htons(UFTRACE_MSG_SEND_META_DATA),
-		.len   = htonl(sizeof(namelen) + namelen),
+		.len   = sizeof(namelen) + namelen,
 	};
 	struct iovec iov[4] = {
 		{ .iov_base = &msg,     .iov_len = sizeof(msg), },
@@ -189,7 +189,7 @@ void send_trace_metadata(int sock, const char *dirname, char *filename)
 	len = stbuf.st_size;
 	buf = xmalloc(len);
 
-	msg.len += htonl(len);
+	msg.len = htonl(msg.len + len);
 	iov[3].iov_base = buf;
 	iov[3].iov_len  = len;
 
